lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -2303,21 +2303,18 @@ func (d *zfs) readonlySnapshot(vol Volume) (string, revert.Hook, error) {
 	d.logger.Debug("Created backup snapshot", logger.Ctx{"dev": srcSnapshot})
 
 	if d.isBlockBacked(vol) {
-		// For block devices, we make them appear.
-		// Check if already active.
-		current, err := d.getDatasetProperty(dataset, "volmode")
+		snapdevMode, err := d.getDatasetProperty(dataset, "snapdev")
```

Comment:
I'd like to understand the why behind this change? Previously it had a comment which seemed to make sense, so im wondering what the issue was here?